OsmAnd/src/net/osmand/plus/routing/data/AnnounceTimeDistances.java

```diff
@@ -174,28 +183,15 @@ public boolean isTurnStateNotPassed(float currentSpeed, double dist, int turnTyp
 	}
 
 	private boolean isDistanceLess(float currentSpeed, double dist, double leadDist) {
-		return isDistanceLess(currentSpeed, dist, leadDist, (float) leadDist / DEFAULT_SPEED);
-	}
-
-	private boolean isDistanceLess(float currentSpeed, double dist, double leadDist, float leadTime) {
-		// Check triggers:
-		// (1) distance <= leadDistance?
-		if (dist <= leadDist + currentSpeed * voicePromptDelayTimeSec) {
-			return true;
-		}
-		// (2) time_with_current_speed <= leadTime? (high speed adjustment)
-		// currentSpeed = 0 in cases where no time threshold is used.
-		return dist <= currentSpeed * (leadTime + voicePromptDelayTimeSec);
+		// Check trigger. Lead distance is scaled up for high speeds. (For cases without such scaling we pass currentSpeed=0.)
+		//return dist <= Math.max(1, currentSpeed / DEFAULT_SPEED) * leadDist + currentSpeed * voicePromptDelayTimeSec;   with division avoided:
+		return dist * DEFAULT_SPEED <= Math.max(DEFAULT_SPEED, currentSpeed) * leadDist + currentSpeed * voicePromptDelayTimeSec * DEFAULT_SPEED;
```

Comment:
Something to take a break may be... 
```
+ currentSpeed * voicePromptDelayTimeSec * DEFAULT_SPEED;
```
This is clearly a typo, and it looks like 2 parts of equastion could be divided by DEFAULT_SPEED to get it simpler

```
dist <= Math.max(distToTrigger, currentSpeed / DEFAULT_SPEED * distToTrigger)  + currentSpeed * voicePromptDelayTimeSec;
```
Naming
dist -> leadDist, leftDist
leadDist -> distanceTrigger

Reply:
I don't see a typo here? The 'simple' equation is the one in the comment in line 187. But it involves a costly float division, I avoid that by multiolying both sides by DEFAULT_SPEED ?